daml-lf/parser/src/main/scala/com/digitalasset/daml/lf/testing/parser/ExprParser.scala

```diff
@@ -149,14 +149,15 @@ private[parser] class ExprParser[P](parserParameters: ParserParameters[P]) {
     }
 
   private lazy val eVariantOrEnumCon: Parser[Expr] =
-    fullIdentifier ~ (`:` ~> id) ~ rep(argTyp) ~ opt(expr0) ^^ {
-      case tName ~ vName ~ argsTyp ~ Some(arg) =>
-        EVariantCon(TypeConApp(tName, argsTyp.to(ImmArray)), vName, arg)
-      case _ ~ _ ~ argsTyp ~ None if argsTyp.nonEmpty =>
-        throw new java.lang.Error("enum type do not take type parameters")
-      case tName ~ vName ~ _ ~ None =>
-        EEnumCon(tName, vName)
-    }
+    phrase(fullIdentifier ~ (`:` ~> id) ~ rep1(argTyp)) ~> err(
```

Comment:
I do not think it is working.
I think you will accept something like `(Mod:R:C" @Bool)` as an enum. 
My intuition is you just cannot put the enum error case before the variant.

Reply:
Slightly reverted code to use `ParsingError` rather than throwing the `java.lang.Error` as we originally did.

Added in specific enum test cases (success and failing) to ensure we have the behaviour we desire here.